src/orch/manager.c

```diff
@@ -0,0 +1,232 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "libhirte/bus/peer-bus.h"
+#include "libhirte/bus/user-bus.h"
+#include "libhirte/common/common.h"
+#include "libhirte/common/parse-util.h"
+#include "libhirte/ini/config.h"
+#include "libhirte/service/shutdown.h"
+#include "libhirte/socket.h"
+
+#include "managednode.h"
+#include "manager.h"
+
+Manager *manager_new(void) {
+        fprintf(stdout, "Creating Manager...\n");
+
+        int r = 0;
+        _cleanup_sd_event_ sd_event *event = NULL;
+        r = sd_event_default(&event);
+        if (r < 0) {
+                fprintf(stderr, "Failed to create event loop: %s\n", strerror(-r));
+                return NULL;
+        }
+
+        _cleanup_free_ char *service_name = strdup(HIRTE_DBUS_NAME);
+        if (service_name == NULL) {
+                fprintf(stderr, "Out of memory\n");
+                return NULL;
+        }
+
+        Manager *manager = malloc0(sizeof(Manager));
+        if (manager != NULL) {
+                manager->port = HIRTE_DEFAULT_PORT;
+                manager->user_bus_service_name = steal_pointer(&service_name);
+                manager->event = steal_pointer(&event);
+                LIST_HEAD_INIT(manager->nodes);
+        }
+
+        return manager;
+}
+
+Manager *manager_ref(Manager *manager) {
+        manager->ref_count++;
+        return manager;
+}
+
+void manager_unref(Manager *manager) {
+        manager->ref_count--;
+        if (manager->ref_count != 0) {
+                return;
+        }
+
+        if (manager->event != NULL) {
+                sd_event_unrefp(&manager->event);
+        }
+
+        free(manager->user_bus_service_name);
+
+        if (manager->node_connection_source != NULL) {
+                sd_event_source_unrefp(&manager->node_connection_source);
+        }
+        if (manager->user_dbus != NULL) {
+                sd_bus_unrefp(&manager->user_dbus);
+        }
+
+        free(manager);
+}
+
+void manager_unrefp(Manager **managerp) {
+        if (managerp && *managerp) {
+                manager_unref(*managerp);
+                *managerp = NULL;
+        }
+}
+
+ManagedNode *manager_find_node(Manager *manager, const char *name) {
+        ManagedNode *node = NULL;
+
+        LIST_FOREACH(nodes, node, manager->nodes) {
+                if (node->name != NULL && strcmp(node->name, name) == 0) {
```

Comment:
Can we use ```strncmp()``` here?

Reply:
Why would we compare just the start of the name?